src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx

```diff
@@ -66,126 +69,119 @@ function stringToQuerySortOrder(value) {
-    useEffect(() => {
-        setShardQueryOptions({
-            wasLoaded: false,
-            data: undefined,
-        });
-    }, [currentSchemaPath, path, setShardQueryOptions]);
+    const {
+        loading,
+        data: {result: data = undefined} = {},
+        error,
+        wasLoaded,
+    } = useTypedSelector((state) => state.shardsWorkload);
+
+    const [sortOrder, setSortOrder] = useState(tableColumnsNames.CPUCores);
+
+    useAutofetcher(() => {
+        dispatch(
+            sendShardQuery({
+                database: path,
+                path: currentSchemaPath,
+                sortOrder: stringToQuerySortOrder(sortOrder),
+            })
+        );
+    }, [dispatch, currentSchemaPath, path, sortOrder], autorefresh);
 
+    // don't show loader for requests triggered by table sort, only for path change
     useEffect(() => {
-        sendShardQuery({
-            database: path,
-            path: currentSchemaPath,
-            sortOrder: stringToQuerySortOrder(sortOrder),
-        });
-    }, [currentSchemaPath, path, sendShardQuery, sortOrder]);
+        dispatch(
+            setShardQueryOptions({
+                wasLoaded: false,
+                data: undefined,
+            })
+        );
+    }, [dispatch, currentSchemaPath, path]);
 
     const history = useContext(HistoryContext);
 
-    const onSchemaClick = (schemaPath) => {
-        return () => {
-            setCurrentSchemaPath(schemaPath);
-            getSchema({path: schemaPath});
-            history.go(0);
-        };
-    };
-
-    const onSort = (newSortOrder) => {
+    const onSort = (newSortOrder?: SortOrder | SortOrder[]) => {
         // omit information about sort order to disable ASC order, only DESC makes sense for top shards
         // use a string (and not the DataTable default format) to prevent reference change,
         // which would cause an excess state change, to avoid repeating requests
         setSortOrder(dataTableToStringSortOrder(newSortOrder));
     };
 
-    const tableColumns = useMemo(() => {
+    const tableColumns: Column<any>[] = useMemo(() => {
+        const onSchemaClick = (schemaPath: string) => {
+            return () => {
+                dispatch(setCurrentSchemaPath(schemaPath));
+                dispatch(getSchema({path: schemaPath}));
+                history.go(0);
+            };
+        };
+
         return [
             {
                 name: tableColumnsNames.Path,
-                // eslint-disable-next-line
                 render: ({value}) => {
                     return (
                         <span
                             // tenant name is substringed out in sql query but is needed here
                             onClick={onSchemaClick(path + value)}
                             className={bLink({view: 'normal'})}
                         >
-                            {value}
+                            {value as string}
```

Comment:
better to have a data shape, but we don't have it yet, probably will introduce it in the next PRs